groovy/postInit/chemistry/organic_chemistry/BiofuelChain.groovy

```diff
@@ -0,0 +1,310 @@
+                .inputs(ore('dustSodiumHydroxide'))
+                .fluidInputs(lipid.get())
+                .fluidInputs(alcohol.get(6000)) // Optimal condition of 6 eq. alcohol
+                .fluidOutputs(alcohol.getGlycerolSolution(3500))
+                .fluidOutputs(alcohol.getCrudeEster(3500))
+                .duration(tpbb * 3)
+                .EUt(30)
+                .buildAndRegister()
+        }
+
+        // Vacuum Flash Distillation of Glycerol Solutions
+        VACUUM_CHAMBER.recipeBuilder()
+            .fluidInputs(fluid('diluted_hydrochloric_acid') * 1000)
+            .fluidInputs(alcohol.getGlycerolSolution(7000))
+            .fluidOutputs(fluid('crude_glycerol') * 3000)
+            .fluidOutputs(alcohol.get(5000))
+            .duration(60)
+            .EUt(30)
+            .buildAndRegister()
+
+        // Distillation of Crude Glycerol
+        DT.recipeBuilder()
+            .fluidInputs(fluid('crude_glycerol') * 3000)
+            .outputs(metaitem('dustSalt'))
+            .fluidOutputs(fluid('glycerol') * 2000)
+            .fluidOutputs(fluid('water') * 1000)
+            .duration(20)
+            .EUt(30)
+            .buildAndRegister()
+
+        // Neutralization of Crude FAXEs
+        MIXER.recipeBuilder()
+            .fluidInputs(fluid('diluted_hydrochloric_acid') * 100)
+            .fluidInputs(alcohol.getCrudeEster(2100))
+            .fluidOutputs(alcohol.getNeutralizedEster(2100))
+            .fluidOutputs(fluid('diluted_saltwater') * 100)
+            .duration(15)
+            .EUt(7)
+            .buildAndRegister()
+
+        // Distillation of Neutralized FAXEs
+        DT.recipeBuilder()
+            .fluidInputs(alcohol.getNeutralizedEster(3500))
+            .fluidOutputs(fluid('unscrubbed_bio_diesel') * 3000)
+            .fluidOutputs(alcohol.get(500))
+            .duration(tpbb * 3)
+            .EUt(30)
+            .buildAndRegister()
+
+        // Washing of Bio Diesel with heated water
+        MIXER.recipeBuilder()
+            .fluidInputs(fluid('gtfo_heated_water') * 1000)
+            .fluidInputs(fluid('unscrubbed_bio_diesel') * 3000)
+            .fluidOutputs(fluid('wastewater') * 1000)
+            .fluidOutputs(fluid('moist_bio_diesel') * 3000)
+            .duration(tpbb * 3)
+            .EUt(7)
+            .buildAndRegister()
+
+        // Drying
+        DRYER.recipeBuilder()
+            .fluidInputs(fluid('moist_bio_diesel') * 1000)
+            .fluidOutputs(fluid('bio_diesel') * 1000)
+            .duration(tpbb)
+            .EUt(30)
+            .buildAndRegister()
+}
+
+// tick per 1/4 bucket of n-paraffin
+int tpbnp = 20
+
+// -------- Green Diesel Chain --------
+// Pretreatment to remove undesired pigments
+// such as carotenoids and chlorophyll via decolorization
+grade1Lipids.forEach { lipid ->
+        MIXER.recipeBuilder()
+            .inputs(ore('dustAluminiumSilicate'))
+            .notConsumable(fluid('phosphoric_acid') * 500)
```

Comment:
im thinking this cost may be a little high

Reply:
do you mean energy? the aluminum silicate itself is recycled in the next step

Reply:
i mean the phosphoric acid